packages/forklift-console-plugin/src/modules/Providers/views/details/components/DetailsSection/utils/getVSphereProviderWebUILink.ts

```diff
@@ -14,13 +14,11 @@ export const getVSphereProviderWebUILink = (provider: V1beta1Provider): string =
   }
 
   // Calculate link using API URL
-  const url = provider?.spec?.url || '';
-  const suffix = '/sdk';
   const newSuffix = '/ui';
+  const uiLinkRegexp = new RegExp('/sdk(/)*$', 'g');
+  const regexpResult = uiLinkRegexp.exec(provider?.spec?.url);
 
-  if (url.endsWith(suffix)) {
-    return url.slice(0, -suffix.length) + newSuffix;
-  } else {
-    return url;
-  }
+  return provider?.spec?.url && regexpResult
```

Comment:
you are replacing an `if...else` expression with an expresion with `?` optional chaining `&&` a `?` trinary opreation and some regex, please revert to the `if else` implementation.

Reply:
The original `if else` doesn't cover all cases (e.g. ending with a valid `/sdk/`)
If url is empty and if /sdk[/] doesn't exist  then we can't auto calculate.

Reply:
the original if else cover `/sdk` suffix and that is the only thing it needs to support, any combintaion with a different suffix does not make sense in this scenario, my concern is that the code is complex and unreadable without a good reason.

Reply:
my concern is the complexity of this new code, using regex and complex logic, when we can do what we need using if...else , it you are concerned about supporting `/sdk/`we can solve it in ways that keep the code simple, for example removing trailing `/` before testing for `/sdk` or testing for both `/sdk` and `/sdk/` the current suggested code is paying too high price in complexity (readability and complexity) to achieve to small pay ( supporting `/sdk/`)

Reply:
OK, sure. The vsphere case is indeed simple so we can achieve the same result without `regexp`.
I'll change it back to use `if..else` with validating `sdk/` as well.

Reply:
After discussing this offline, we will remove the validation of the URL path with a trailing slash for both RHV and vSphere (e.g. `sdk/, api/`) and therefore remove the auto calculation for this as well.